### ISSUE TITLE
Rename ESNIKeys to something more generic

### ISSUE THREAD

**Issue by ghedo** [ekr/draft-rescorla-tls-esni#30]
With #22 merged, I wonder if the ESNIKeys struct should just be renamed to something not specific to ESNI, but something more like "generic TLS config in DNS" so that it can be more cleanly reused by unrelated features.

"ServerConfiguration" comes to mind from the early TLS 1.3 days.

**Comment by chris-wood** [comment]
If we were to re-use these keys for other purposes, that seems fine. However, I'm not sure we want to do that. Semi-static keys, for example, should ideally be separate.

**Comment by ghedo** [comment]
The keys yes, but I imagine the extensions field can hold "arbitrary" data, so while the keys would be ESNI-specific, you could also include extensions that have nothing to do with ESNI.

**Comment by ghedo** [comment]
Thinking about this some more, this would also require changing the "_esni." prefix to something else.

In any case I can see how this might be out of scope for this spec. In the end it's a matter of deciding whether future TLS extensions that need similar DNS records, should be able to reuse the same structure (so no additional TXT record with its own prefix would be required), or if they should define their own structure (and have a separate TXT record).

I'm happy to close this if it goes too much out of scope.

**Comment by chris-wood** [comment]
Right, I think it's verging on being out of scope. What do others think? @kazuho @ekr?

**Comment by kazuho** [comment]
My preference goes to keeping the name as-is, considering the fact that the role of ESNIKeys is  (at least for the moment) to negotiate properties between the client and the fronting server (not the hidden server), and that the only property we need to negotiate between the two is the information necessary for ESNI protection.

It is true that the DNS record can covey properties related to the hidden server (as you know, I've argued for using it to carry the server certificate chain). But IMO that's a change of concept, and I prefer keeping the concept simple for the time being.

**Comment by ghedo** [comment]
Ok.
